go-controller/pkg/ovn/controller/egressfirewall/egressfirewall.go

```diff
@@ -414,9 +415,58 @@ func (oc *EFController) addEgressFirewall(egressFirewall *egressfirewallapi.Egre
 	return nil
 }
 
+// validateAndGetEgressFirewallDestination validates an egress firewall rule destination and returns
+// the parsed contents of the destination.
+func (oc *EFController) validateAndGetEgressFirewallDestination(namespace string, egressFirewallDestination egressfirewallapi.EgressFirewallDestination) (
+	cidrSelector string,
+	dnsName string,
+	clusterSubnetIntersection []*net.IPNet,
+	nodeSelector *metav1.LabelSelector,
+	err error) {
+	// Validate the egress firewall rule.
+	if egressFirewallDestination.DNSName != "" {
+		n, err := util.ValidateAndGetEgressFirewallDNSDestination(egressFirewallDestination)
+		if err != nil {
+			return "", "", nil, nil, err
+		}
+		dnsName = n
+	} else if len(egressFirewallDestination.CIDRSelector) > 0 {
+		// Validate CIDR selector.
+		_, ipNet, err := net.ParseCIDR(egressFirewallDestination.CIDRSelector)
+		if err != nil {
+			return "", "", nil, nil, err
+		}
+		cidrSelector = egressFirewallDestination.CIDRSelector
+		netInfo, err := oc.networkManager.GetActiveNetworkForNamespace(namespace)
+		if err != nil {
+			return "", "", nil, nil,
+				fmt.Errorf("failed to validate egress firewall destination: %w", err)
+		}
+		subnets := config.Default.ClusterSubnets
+		if !netInfo.IsDefault() {
+			subnets = netInfo.Subnets()
+		}
+		for _, clusterSubnet := range subnets {
+			if clusterSubnet.CIDR.Contains(ipNet.IP) || ipNet.Contains(clusterSubnet.CIDR.IP) {
+				clusterSubnetIntersection = append(clusterSubnetIntersection, clusterSubnet.CIDR)
```

Comment:
did you mean to assign this value to the matchTarget?

Reply:
yeah